src/main/java/org/terasology/scenario/components/information/IndentificationComponents/ScenarioTypeIntegerComponent.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.scenario.components.information.IndentificationComponents;
+
+import org.terasology.entitySystem.Component;
+import org.terasology.network.Replicate;
+
+/**
+ * Type component for a Scenario argument entity, indicates that the argument is an integer typed value
+ *
+ * Argument entities include:
+ *   Network Component
+ *   Type Component

```

Comment:
I think in general {@link } would be better than just saying "Type Component" as it is probably now not called TypeComponent.